semgrep-core/src/core/Semgrep_error_code.ml

```diff
@@ -0,0 +1,247 @@
+      let loc = Parse_info.first_loc_of_file file in
+      let msg = Common.string_of_timeout_info timeout_info in
+      mk_error_no_rule loc msg Timeout
+  | Out_of_memory ->
+      let loc = Parse_info.first_loc_of_file file in
+      mk_error_no_rule loc "Out of memory" OutOfMemory
+  | UnixExit _ as exn -> raise exn
+  (* general case, can't extract line information from it, default to line 1 *)
+  | exn ->
+      let loc = Parse_info.first_loc_of_file file in
+      {
+        rule_id = None;
+        typ = FatalError;
+        loc;
+        msg = Common.exn_to_s exn;
+        details = Some (Printexc.get_backtrace ());
+        yaml_path = None;
+      }
+
+(*****************************************************************************)
+(* Pretty printers *)
+(*****************************************************************************)
+
+let string_of_error err =
+  let pos = err.loc in
+  assert (pos.PI.file <> "");
+  spf "%s:%d:%d: %s" pos.PI.file pos.PI.line pos.PI.column err.msg
+
+let string_of_error_kind = function
+  | LexicalError -> "Lexical error"
+  | ParseError -> "Syntax error"
+  | SpecifiedParseError -> "Other syntax error"
+  | AstBuilderError -> "AST builder error"
+  (* pattern parsing related errors *)
+  | RuleParseError -> "Rule parse error"
+  | PatternParseError -> "Pattern parse error"
+  | InvalidYaml -> "Invalid YAML"
+  (* semgrep *)
+  | SemgrepMatchFound check_id -> spf "sgrep-lint-<%s>" check_id
+  | MatchingError -> "Internal matching error"
+  | TooManyMatches -> "Too many matches"
+  (* other *)
+  | FatalError -> "Fatal error"
+  | Timeout -> "Timeout"
+  | OutOfMemory -> "Out of memory"
+
+let severity_of_error typ =
+  match typ with
+  | SemgrepMatchFound _title -> Warning
+  | MatchingError -> Warning
+  | TooManyMatches -> Warning
+  | LexicalError -> Warning
+  | ParseError -> Warning
+  | SpecifiedParseError -> Warning
+  | AstBuilderError -> Error
+  | RuleParseError -> Error
+  | PatternParseError -> Error
+  | InvalidYaml -> Warning
+  | FatalError -> Error
+  | Timeout -> Warning
+  | OutOfMemory -> Warning
+
+(*****************************************************************************)
+(* Try with error *)
+(*****************************************************************************)
+
+let try_with_exn_to_error file f =
+  try f () with exn -> Common.push (exn_to_error file exn) g_errors
+
+let try_with_print_exn_and_reraise file f =
+  try f ()
+  with exn ->
+    let bt = Printexc.get_backtrace () in
+    let err = exn_to_error file exn in
+    pr2 (string_of_error err);
+    pr2 bt;
+    (* does not really re-raise :( lose some backtrace *)
+    raise exn
+
+(* fast = no stack trace *)
+let try_with_print_exn_and_exit_fast file f =
+  try f ()
+  with exn ->
+    let err = exn_to_error file exn in
+    pr2 (string_of_error err);
+    exit 2
+
+(*****************************************************************************)
+(* Helper functions to use in testing code *)
```

Comment:
This doesn't seem very related to error codes?

Reply:
It was in Error_code.ml, but you're right, we should probably separate that out.